packages/protocol/lib/fed-attestations-utils.ts

```diff
@@ -0,0 +1,80 @@
+import { ensureLeading0x } from '@celo/base'
+import { Address } from '@celo/utils/lib/address'
+import { structHash } from '@celo/utils/lib/sign-typed-data-utils'
+import { generateTypedDataHash } from '@celo/utils/src/sign-typed-data-utils'
+import { parseSignatureWithoutPrefix } from '@celo/utils/src/signatureUtils'
+
+export interface AttestationDetails{
+  identifier: string,
+  issuer: string,
+  account: string,
+  issuedOn: number,
+}
+
+const getTypedData = (chainId: number, contractAddress: Address, message?: AttestationDetails) => {
+  const typedData =  {
+    types: {
+      EIP712Domain: [
+        { name: 'name', type: 'string' },
+        { name: 'version', type: 'string' },
+        { name: 'chainId', type: 'uint256'}, 
+        { name: 'verifyingContract', type: 'address'}, 
+      ],
+      OwnershipAttestation: [
+          { name: 'identifier', type: 'bytes32' },
+          { name: 'issuer', type: 'address'},
+          { name: 'account', type: 'address' },
+          { name: 'issuedOn', type: 'uint256' },
+          // TODO ASv2 Consider including a nonce (which could also be used as an ID)
+      ],
+    },
+    primaryType: 'OwnershipAttestation',
+    domain: {
+      name: 'FederatedAttestations',
+      version: '1.0',
+      chainId,
+      verifyingContract: contractAddress
+    },
+    message: message ? message : {}
+  }
+  return typedData
+}
+
+export const getSignatureForAttestation = async (
+  identifier: string,
+  issuer: string,
+  account: string,
+  issuedOn: number,
+  signer: string,
+  chainId: number,
+  contractAddress: string
+) => {
+  const typedData = getTypedData(chainId, contractAddress, { identifier,issuer,account, issuedOn})
+
+  const signature = await new Promise<string>((resolve, reject) => {
+    web3.currentProvider.send(
```

Comment:
Where is web3 defined here?

Reply:
cc @isabellewei  who can probably speak to this more, but I think this is following how this is done in `meta-tx-utils`, and looks like it ends up being the `web3` object made available by truffle when running tests 🤔

Reply:
yes I think truffle does some behind the scenes magic and declares it before running the tests